### PR TITLE
Add option to exclude library symbols from navTo results

### LINKED ISSUE #48194: Add option for navto to exclude built-in libs and files from node_modules
# Feature request

### 🔎 Search Terms
- navto
- go to symbol in workspace
- tsserver

### Request
Currently `navto` includes all known symbols in a project. This include both symbols from the stdlib and from `node_modules`. However the most common use case for `navto` in my experience is jumping to another symbol within my code. 

I propose we add new options to `navto` that let us exclude:

- Symbols from the standard library
- Any symbols from `node_modules`

This could be exposed as either a single new option on the request that's called something like `justMyCode`, or as two separate options for more fine grained control. In either case, we would expose the option(s) to users in the VS Code settings UI

Extracted https://github.com/microsoft/vscode/issues/46718 as the most common use case there seems to be excluding files under `node_modules`

### PR DESCRIPTION
**Pull request by gabritto**
Fixes #48194.

I'll follow up with an equivalent setting in vscode.

### CODE REVIEW

File: src/harness/client.ts [hunk 1]
```diff
@@ -579,14 +589,13 @@ export class SessionClient implements LanguageService {
             const providePrefixAndSuffixTextForRename = typeof preferences === "boolean" ? preferences : preferences?.providePrefixAndSuffixTextForRename;
             const quotePreference = typeof preferences === "boolean" ? undefined : preferences?.quotePreference;
             if (providePrefixAndSuffixTextForRename !== undefined || quotePreference !== undefined) {
+                const oldPreferences = this.preferences;
```
Comment (gabritto) [inline src/harness/client.ts:592]: The other diffs in the file were just a drive-by fix to how we restore preferences for the harness client. I think we don't have tests that observe this though.

File: src/compiler/types.ts [hunk 2]
```diff
@@ -9936,6 +9936,7 @@ export interface UserPreferences {
     readonly organizeImportsNumericCollation?: boolean;
     readonly organizeImportsAccentCollation?: boolean;
     readonly organizeImportsCaseFirst?: "upper" | "lower" | false;
+    readonly excludeLibrarySymbols?: boolean;
```
Comment (andrewbranch) [inline src/compiler/types.ts:9939]: I think the name of this setting needs to mention the command/feature it affects.

File: src/harness/client.ts [hunk 3]
```diff
@@ -314,15 +314,25 @@ export class SessionClient implements LanguageService {
         return notImplemented();
     }
 
-    getNavigateToItems(searchValue: string): NavigateToItem[] {
+    getNavigateToItems(searchValue: string, maxResultCount: number, file: string | undefined, _excludeDtsFiles: boolean | undefined, excludeLibFiles: boolean | undefined): NavigateToItem[] {
         const args: protocol.NavtoRequestArgs = {
             searchValue,
-            file: this.host.getScriptFileNames()[0],
+            file,
+            currentFileOnly: !!file,
```
Comment (andrewbranch) [inline src/harness/client.ts:321]: One thought: if I have `excludeLibrarySymbols` enabled, but then I manually open `node_modules/typescript/lib/lib.dom.d.ts` and use ⌘⇧O (symbol search in current file), I still want to see results from the current file.

**Review by andrewbranch — APPROVED** [review]
